### PR TITLE
sanitize column names for csv

### PR DESCRIPTION
**Pull request by PangZhi**
### Description and related tickets, documents
Sanitize column names when loading csv

Replace invalid characters by dash _
Remove invalid beginning, e.g: _a_b_c__d_ => a_b_c__d, 1a => a
If encounter duplicated column names after above steps, add number at the end: a_b, a_b, a_b => a_b, a_b_1, a_b_2

Reviewers: @Huandao0812 @nhanitvn @hai-adatao 

### How to test
UT, client test

### PR Progress
Make sure all checkboxes below are checked before merged
- [x] Branch is in format `prefix/description` (see [this](http://www.guyroutledge.co.uk/blog/git-branch-naming-conventions/))
- [x] Merge check has no conflicts. PR checks passed.
- [ ] Code review is done.

### CODE REVIEW

File: core/src/main/java/io/ddf/DDFManager.java [hunk 1]
```diff
@@ -433,9 +433,7 @@ private DDF newDDF(Class<?>[] argTypes, Object[] argValues) throws DDFException
       return ddf;
 
     } catch (Exception e) {
-      throw new  DDFException(String.format(
```
Comment (hai-adatao) [inline core/src/main/java/io/ddf/DDFManager.java]: Why do we remove this error message?

File: spark/src/test/java/io/ddf/spark/SparkDDFManagerTests.java [hunk 2]
```diff
@@ -125,9 +129,37 @@ public void testBasicCopyForS3(S3DDFManager s3DDFManager) throws DDFException {
...
+        .add("dd2_2_2")
+        .add("dd_22")
+        .add("dd_22_0")
+        .build();
+
+    StringBuilder schema = new StringBuilder(String.format("%s int", nameList.get(0)));
+    for (int i = 1; i < nameList.size(); ++i) {
+      schema.append(String.format(", %s int", nameList.get(i)));
+    }
+    S3DDF ddf = s3DDFManager.newDDF("adatao-sample-data", "test/csv/6col.csv", schema.toString(), null);
```
Comment (hai-adatao) [inline spark/src/test/java/io/ddf/spark/SparkDDFManagerTests.java]: Please help put this file under `adatao-sample-data/test/pa/<folder>/<filename>`, the reason for this is that other part of this `test` folder usually get deleted and upload again using a zip file, so if putting it under `csv`, this file may get deleted

File: core/src/main/java/io/ddf/DDFManager.java (hunk 1 above)
Comment (nhanitvn) [inline core/src/main/java/io/ddf/DDFManager.java]: To not hide and properly bubble up the underlying error.

File: core/src/main/java/io/ddf/util/Utils.java [hunk 3]
```diff
@@ -66,6 +69,46 @@ public static DataFormat getDataFormatFromPath(String path) throws DDFException
...
 
+
+  public static List<String> sanitizeColumnName(List<String> columnNames) throws DDFException {
+    Preconditions.checkArgument(columnNames.size() > 0);
+    List<String> ret = new ArrayList<>();
+    for (String name : columnNames) {
+      Preconditions.checkArgument(name.length() > 0);
+      String dashName = name.trim().replaceAll("![0-9a-zA-Z]|\\s", "_");
+      int i;
+      for (i = 0; i < dashName.length(); ++i) {
```
Comment (nhanitvn) [inline core/src/main/java/io/ddf/util/Utils.java]: Why do we need this code piece given that we replaced all non alpha-numeric chars with _? @PangZhi

File: core/src/main/java/io/ddf/util/Utils.java [hunk 4]
```diff
@@ -66,6 +69,46 @@ public static DataFormat getDataFormatFromPath(String path) throws DDFException
...
+          ret.add(dashName.substring(i));
+          break;
+        }
+      }
+      if (i == dashName.length()) {
+        // TODO: get some random name
+        throw new DDFException(String.format("Not valid column name: %s", name));
+      }
+    }
+    HashSet<String> nameSet = new HashSet<>();
```
Comment (nhanitvn) [inline core/src/main/java/io/ddf/util/Utils.java]: Should add comment to describe the purpose of the following code. It is for avoiding duplicated column names, right?

Comment (hai-adatao) [inline core/src/main/java/io/ddf/util/Utils.java]: Right, not just this piece of code, it's good if we can document what this function does in the beginning of this function.

File: core/src/main/java/io/ddf/util/Utils.java (hunk 3 above)
Comment (PangZhi) [inline core/src/main/java/io/ddf/util/Utils.java]: @nhanitvn @hai-adatao This is to replace space and non-valid characters such as "@" in the column names

**Comment by Huandao0812** [comment]
lgtm

**Pull request merged**